test-poc/framework/src/main/java/org/keycloak/test/framework/server/KeycloakTestServerSupplier.java

```diff
@@ -0,0 +1,50 @@
+package org.keycloak.test.framework.server;
+
+import org.keycloak.test.framework.KeycloakIntegrationTest;
+import org.keycloak.test.framework.injection.InstanceWrapper;
+import org.keycloak.test.framework.injection.LifeCycle;
+import org.keycloak.test.framework.injection.Registry;
+import org.keycloak.test.framework.injection.Supplier;
+import org.keycloak.test.framework.injection.SupplierHelpers;
+
+public class KeycloakTestServerSupplier implements Supplier<KeycloakTestServer, KeycloakIntegrationTest> {
+
+    @Override
+    public Class<KeycloakTestServer> getValueType() {
+        return KeycloakTestServer.class;
+    }
+
+    @Override
+    public Class<KeycloakIntegrationTest> getAnnotationClass() {
+        return KeycloakIntegrationTest.class;
+    }
+
+    @Override
+    public InstanceWrapper<KeycloakTestServer, KeycloakIntegrationTest> getValue(Registry registry, KeycloakIntegrationTest annotation) {
+        KeycloakTestServerConfig serverConfig = SupplierHelpers.getInstance(annotation.config());
+
+//        RemoteKeycloakTestServer keycloakTestServer = new RemoteKeycloakTestServer();
+        EmbeddedKeycloakTestServer keycloakTestServer = new EmbeddedKeycloakTestServer();
```

Comment:
What is the plan for switching between these in the future? Do we have an issue for this?

Reply:
Yes, @lhanusov is working on it

Reply:
Yeah saw it in the meeting, we'll deal with this after